### PR TITLE
BUG: fix constraints and bunds when use scipy.optimize.minimize

### LINKED ISSUE #6655: got multiple values for argument 'constraints', 'bounds'
#### Describe the bug
error happens when provide "bounds" or "constraints" option for scipy minimize

#### Code Sample, a copy-pastable example if possible

```python
~/anaconda3/lib/python3.7/site-packages/statsmodels/base/optimizer.py in _fit_minimize(f, score, start_params, fargs, kwargs, disp, maxiter, callback, retall, full_output, hess)
    273 
    274     res = optimize.minimize(f, start_params, args=fargs, method=kwargs['min_method'],
--> 275                             jac=score, hess=hess, callback=callback, options=options)
    276 
    277     xopt    = res.x

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_minimize.py in minimize(fun, x0, args, method, jac, hess, hessp, bounds, constraints, tol, callback, options)
    616     elif meth == 'slsqp':
    617         return _minimize_slsqp(fun, x0, args, jac, bounds,
--> 618                                constraints, callback=callback, **options)
    619     elif meth == 'trust-constr':
    620         return _minimize_trustregion_constr(fun, x0, args, jac, hess, hessp,

TypeError: _minimize_slsqp() got multiple values for argument 'constraints'

```
<details>

This happens when a keyword argument is specified that overwrites a positional argument. 
as scipy minimize already has bounds, constraints in the signature, and the options pass the bunds and constraints again. 

It should handled the same way as jac and hess, thank you
</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Expected Output

Run without give error

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>
statsmodels
===========

Installed: 0.11.1 
</details>

### PR DESCRIPTION
**Pull request by emailhy**
- [x] closes #6655 
- [x] tests added / passed. 
- [x] code/documentation is well formatted.  
- [x] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by emailhy** [comment]
this should fix issue 6655

**Comment by josef-pkt** [comment]
The travis failure is unrelated to this PR (connection failure when retrieving an rdataset)

File: statsmodels/base/optimizer.py [hunk 1]
```diff
@@ -271,8 +271,22 @@ def _fit_minimize(f, score, start_params, fargs, kwargs, disp=True,
...
+        bounds = None
+
+    if 'constraints' in kwargs.keys() and kwargs['min_method'] in has_constraints:
+        constraints = kwargs['constraints']
+    else:
+        constraints = ()
+
     res = optimize.minimize(f, start_params, args=fargs, method=kwargs['min_method'],
-                            jac=score, hess=hess, callback=callback, options=options)
+                            jac=score, hess=hess, bounds=bounds, constraints=constraints, callback=callback, options=options)
```
Comment (josef-pkt) [inline statsmodels/base/optimizer.py:289]: line too long

File: statsmodels/base/tests/test_optimize.py [hunk 2]
```diff
@@ -75,3 +89,10 @@ def test_full_output(reset_randomstate):
             assert_(xopt.shape == () and xopt.size == 1)
         else:
             assert_(len(xopt) == 1)
+
+def test_minimize_scipy_slsqp():
+    func =fit_funcs['minimize']
```
Comment (josef-pkt) [inline statsmodels/base/tests/test_optimize.py:94]: space around `=`

File: statsmodels/base/tests/test_optimize.py [hunk 3]
```diff
@@ -75,3 +89,10 @@ def test_full_output(reset_randomstate):
...
+    xopt, retvals = func(dummy_bounds_constraint_func, None, (2,0), (), {'min_method':'SLSQP',
+                    'bounds':dummy_bounds(), 'constraints':dummy_constraints()},  hess=None, full_output=False, disp=0)
```
Comment (josef-pkt) [inline statsmodels/base/tests/test_optimize.py:96]: lines too long

**Comment by josef-pkt** [comment]
Thanks, LGTM

I check where we could add a unit test that uses a `model.fit`

File: statsmodels/base/tests/test_optimize.py (hunk 3 above)
Comment (emailhy) [reply statsmodels/base/tests/test_optimize.py:96]: fixed in the new commit

File: statsmodels/base/tests/test_optimize.py (hunk 2 above)
Comment (emailhy) [reply statsmodels/base/tests/test_optimize.py]: fixed in the new commit.

File: statsmodels/base/optimizer.py (hunk 1 above)
Comment (emailhy) [reply statsmodels/base/optimizer.py]: fixed in the new commit

**Comment by josef-pkt** [comment]
I didn't have time to try out an example.

However, it looks good and can be merged.

Can you squash the commits given that most commits are pep-8 and not informative in themselves?

**Comment by josef-pkt** [comment]
@emailhy thanks and sorry for the delay.

ready to merge when CI comes back green

**Comment by emailhy** [comment]
> 
> 
> I didn't have time to try out an example.
> 
> However, it looks good and can be merged.
> 
> Can you squash the commits given that most commits are pep-8 and not informative in themselves?

Thank you and I squashed all commits.

**Comment by josef-pkt** [comment]
all green, merging

@emailhy thanks again for the fix

**Pull request merged**